interface/Tools.h

```diff
@@ -74,6 +76,267 @@ MELAAngles HHAnalyzer::getMELAAngles(const LorentzVector &q1, const LorentzVecto
+    std::vector<std::string> filter_leg2;
+    if (leptons[dilepton.ilep1].isMu && leptons[dilepton.ilep2].isMu) {
+        // di-muon filters: from the path name the legs can have asymetric cuts, taken filters from 
+        // https://github.com/cms-analysis/MuonAnalysis-TagAndProbe/blob/fa1f8f3d469a5a78754ed4b4c43adbfad39a2544/python/common_variables_cff.py#L253-L264
+        if (HH_HLT_DEBUG) std::cout << "\tfinding dilepton legs: di-muon" << std::endl;
+        filter_leg1.push_back("hltL3fL1sDoubleMu114L1f0L2f10OneMuL3Filtered17");
+        filter_leg1.push_back("hltL3fL1sDoubleMu114L1f0L2f10L3Filtered17");
+        filter_leg2.push_back("hltL3pfL1sDoubleMu114L1f0L2pf0L3PreFiltered8");
+        filter_leg2.push_back("hltDiMuonGlbFiltered17TrkFiltered8");
+        // di-electron case: from the path name the legs have symetric cuts, do not check on filters
+        // different flavour case: we can match the flavour at HLT, so we have a clear leg1 and leg2
+        for (auto& i1: l1_samepath_indices)
+            for (auto& f1: filter_leg1)
+                if (std::find(hlt.object_filters[i1].begin(), hlt.object_filters[i1].end(), f1) != hlt.object_filters[i1].end())
+                    leptons[dilepton.ilep1].hlt_leg1 = true;
+        for (auto& i1: l1_samepath_indices)
+            for (auto& f2: filter_leg2)
+                if (std::find(hlt.object_filters[i1].begin(), hlt.object_filters[i1].end(), f2) != hlt.object_filters[i1].end())
+                    leptons[dilepton.ilep1].hlt_leg2 = true;
+        for (auto& i2: l2_samepath_indices)
+            for (auto& f1: filter_leg1)
+                if (std::find(hlt.object_filters[i2].begin(), hlt.object_filters[i2].end(), f1) != hlt.object_filters[i2].end())
+                    leptons[dilepton.ilep2].hlt_leg1 = true;
+        for (auto& i2: l2_samepath_indices)
+            for (auto& f2: filter_leg2)
+                if (std::find(hlt.object_filters[i2].begin(), hlt.object_filters[i2].end(), f2) != hlt.object_filters[i2].end())
+                    leptons[dilepton.ilep2].hlt_leg2 = true;
+    } else if (leptons[dilepton.ilep1].isEl && leptons[dilepton.ilep2].isEl) {
+        // di-electron filters: from the path name the legs have symetric cuts
+        // meaning: set everything to true, leg1 and leg2 are determined by the highest hlt pt
+        if (HH_HLT_DEBUG) std::cout << "\tfinding dilepton legs: di-electron" << std::endl;
+        leptons[dilepton.ilep1].hlt_leg1 = true;
+        leptons[dilepton.ilep1].hlt_leg2 = true;
+        leptons[dilepton.ilep2].hlt_leg1 = true;
+        leptons[dilepton.ilep2].hlt_leg2 = true;
+    } else if ((leptons[dilepton.ilep1].isEl && leptons[dilepton.ilep2].isMu)
+        || (leptons[dilepton.ilep1].isMu && leptons[dilepton.ilep2].isEl)) {
+        // if the two offline objects are matching the same different flavour HLT path
+        // then the leg1 and leg2 assignment is in sync with the order of the path name itself
```

Comment:
I need to think a bit more about this case, but indeed maybe there's no need here for a in depth leg matching